### PR TITLE
Enable I2C FIFO full hold in slave mode (stretch clock when RX full), fixes #456

### LINKED ISSUE #456: Enable I2C slave clock stretch by default?
I am using the RP2040 as an I2C slave, but it took quite a bit of trial and error before I realized that clock stretching is not enabled by default.

Would it make sense to set `IC_CON.RX_FIFO_FULL_HLD_CTRL` to 1 by default as part of `i2c_set_slave_mode`?

### PR DESCRIPTION
**Pull request by Wren6991**
See #456

### CODE REVIEW

**Comment by kilograham** [comment]
@reynoldsbd does this look good to yoou

File: src/rp2_common/hardware_i2c/i2c.c [hunk 1]
```diff
@@ -115,16 +115,18 @@ void i2c_set_slave_mode(i2c_inst_t *i2c, bool slave, uint8_t addr) {
     invalid_params_if(I2C, addr >= 0x80); // 7-bit addresses
     invalid_params_if(I2C, i2c_reserved_addr(addr));
     i2c->hw->enable = 0;
+    uint32_t ctrl_set_if_master = I2C_IC_CON_MASTER_MODE_BITS | I2C_IC_CON_IC_SLAVE_DISABLE_BITS;
+    uint32_t ctrl_set_if_slave = I2C_IC_CON_RX_FIFO_FULL_HLD_CTRL_BITS;
     if (slave) {
-        hw_clear_bits(&i2c->hw->con,
-                      I2C_IC_CON_MASTER_MODE_BITS |
-                      I2C_IC_CON_IC_SLAVE_DISABLE_BITS
+        hw_write_masked(&i2c->hw->con,
```
Comment (reynoldsbd) [inline src/rp2_common/hardware_i2c/i2c.c:121]: Personally, I found the older form (`hw_clear_bits`/`hw_write_bits` with literal arguments) to be a bit easier to digest

**Comment by reynoldsbd** [comment]
@kilograham @Wren6991 yes I believe this would do the trick, thanks!

**Review by kilograham — COMMENTED** [review]
i believe!

**Review by kilograham — APPROVED** [review]

**Pull request merged**